maketgz: switch to xz instead of lzma

bagder opened the pull request:
The compressed output size seems to be a tad bit smaller, but generally
xz seems more preferred these days and is used directly by for example
gentoo instead of bz2.

"Users of LZMA Utils should move to XZ Utils" =>
https://tukaani.org/lzma/

dfandrich commented:
The daily snapshots stopped being built shortly after this was committed. Coincidence?

And on an unrelated note, the tradeoff for using xz -9 is much worse than for gzip. It drastically increases the amount of RAM required for decompressing, without improving compression for smaller-sized files. The docs say "it is good to avoid using the presets -7 ... -9 when there's no real need for them."

bagder commented:
> Coincidence?

Nope, I managed to break the script when I updated it for the new extension! Fixed now.

> the tradeoff for using xz -9 is much worse than for gzip

I changed it to -6e now, as that compresses almost as good. Thanks!
